### PR TITLE
RPE-1005 Add IdamTestApi

### PR DESCRIPTION
**Pull request by timja**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RPE-1005

### Change description ###
Add test api to make integration / functional testing easier and prevent having to duplicate code all over the place

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/test/UserGroup.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.idam.client.models.test;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public class UserGroup {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/idam/client/models/test/UserGroup.java]: Do you really need this class. Can you not just have `private final String userGroup` in `CreateUserRequest` ?

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/test/UserRole.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.idam.client.models.test;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public class UserRole {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/idam/client/models/test/UserRole.java]: Do you really need this class. Can you not just have `private final List<String> roles ` in `CreateUserRequest` ?

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/test/CreateUserRequest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.List;
+
+@Getter
+@Builder
+public class CreateUserRequest {
+
+    public static final String DEFAULT_PASSWORD = "Password12";
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/idam/client/models/test/CreateUserRequest.java]: Do you need to have default values for some of the variables in this class ?
